# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../clanlib06/directfb-api-change.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur ClanLib-0.6.5-orig/Sources/Display/Input/DirectFB/keyboard_directfb.cpp ClanLib-0.6.5/Sources/Display/Input/DirectFB/keyboard_directfb.cpp
--- ClanLib-0.6.5-orig/Sources/Display/Input/DirectFB/keyboard_directfb.cpp	2002-06-07 19:36:33.000000000 +0000
+++ ClanLib-0.6.5/Sources/Display/Input/DirectFB/keyboard_directfb.cpp	2006-06-04 12:28:34.000000000 +0000
@@ -242,8 +242,7 @@
     case DIKI_SHIFT_L:     return CL_KEY_LSHIFT;
     case DIKI_SHIFT_R:     return CL_KEY_RSHIFT;
     case DIKI_ALT_L:       return CL_KEY_ALT;
-    case DIKI_ALT_R:       return CL_KEY_ALT;
-    case DIKI_ALTGR:       return CL_KEY_ALTGR;
+    case DIKI_ALT_R:       return CL_KEY_ALTGR;
     case DIKI_TAB:         return CL_KEY_TAB;
     case DIKI_ENTER:       return CL_KEY_ENTER;
     case DIKI_SPACE:       return CL_KEY_SPACE;
@@ -337,7 +336,7 @@
     case CL_KEY_LSHIFT:    return DIKI_SHIFT_L;
     case CL_KEY_RSHIFT:    return DIKI_SHIFT_R;
     case CL_KEY_ALT:       return DIKI_ALT_L;
-    case CL_KEY_ALTGR:     return DIKI_ALTGR;
+    case CL_KEY_ALTGR:     return DIKI_ALT_R;
     case CL_KEY_TAB:       return DIKI_TAB;
     case CL_KEY_ENTER:     return DIKI_ENTER;
     case CL_KEY_SPACE:     return DIKI_SPACE;
